core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -550,6 +550,231 @@ public void testDeleteFilesTableSelection() throws IOException {
+            .withPath("/path/to/data-11.parquet")
+            .withRecordCount(10)
+            .withFileSizeInBytes(10)
+            .withPartition(data11Key)
+            .build();
+
+    table.newFastAppend().appendFile(data10).commit();
+    table.newFastAppend().appendFile(data11).commit();
+
+    Table metadataTable = new PartitionsTable(table.ops(), table);
+    Expression filter = Expressions.and(
+            Expressions.equal("partition.id", 10),
+            Expressions.greaterThan("record_count", 0));
+    TableScan scan = metadataTable.newScan().filter(filter);
+    CloseableIterable<FileScanTask> tasks = PartitionsTable.planFiles((StaticTableScan) scan);
+
+    // Four data files of old spec, one new data file of new spec
+    Assert.assertEquals(5, Iterables.size(tasks));
+
+    filter = Expressions.and(
+            Expressions.equal("partition.data_bucket", 0),
+            Expressions.greaterThan("record_count", 0));
+    scan = metadataTable.newScan().filter(filter);
+    tasks = PartitionsTable.planFiles((StaticTableScan) scan);
+
+    // 1 original data file written by old spec, plus 1 new data file written by new spec
+    Assert.assertEquals(2, Iterables.size(tasks));
+  }
+
+  @Test
+  public void testPartitionSpecEvolutionAdditiveV2() {
+    preparePartitionedTable();
+    Assume.assumeTrue(formatVersion == 2);
+
+    // Change spec and add two data and delete files each
+    table.updateSpec()
+            .addField("id")
+            .commit();
+    PartitionSpec newSpec = table.spec();
+
+    // Add two data files and two delete files with new spec
+    DataFile data10 = DataFiles.builder(newSpec)
+            .withPath("/path/to/data-10.parquet")
+            .withRecordCount(10)
+            .withFileSizeInBytes(10)
+            .withPartitionPath("data_bucket=0/id=10")
+            .build();
+    DataFile data11 = DataFiles.builder(newSpec)
+            .withPath("/path/to/data-11.parquet")
+            .withRecordCount(10)
+            .withFileSizeInBytes(10)
+            .withPartitionPath("data_bucket=1/id=11")
+            .build();
+
+    DeleteFile delete10 = FileMetadata.deleteFileBuilder(newSpec)
+            .ofPositionDeletes()
+            .withPath("/path/to/data-10-deletes.parquet")
+            .withFileSizeInBytes(10)
+            .withPartitionPath("data_bucket=0/id=10")
+            .withRecordCount(1)
+            .build();
+    DeleteFile delete11 = FileMetadata.deleteFileBuilder(newSpec)
+            .ofPositionDeletes()
+            .withPath("/path/to/data-11-deletes.parquet")
+            .withFileSizeInBytes(10)
+            .withPartitionPath("data_bucket=1/id=11")
+            .withRecordCount(1)
+            .build();
+
+    table.newFastAppend().appendFile(data10).commit();
+    table.newFastAppend().appendFile(data11).commit();
+
+    if (formatVersion == 2) {
+      table.newRowDelta().addDeletes(delete10).commit();
+      table.newRowDelta().addDeletes(delete11).commit();
+    }
```

Comment:
Nit: Since this test has an assumption that the format is v2, is the if condition needed / does it provide any benefit?

Reply:
Good catch, removed.